test/suites/integration/httproute_update_test.go

```diff
@@ -0,0 +1,144 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/types"
+	"log"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/latticestore"
+	"github.com/aws/aws-application-networking-k8s/test/pkg/test"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+)
+
+var _ = Describe("HTTPRoute Update", func() {
+
+	var (
+		gateway               *v1beta1.Gateway   = nil
+		pathMatchHttpRouteOne *v1beta1.HTTPRoute = nil
+		pathMatchHttpRouteTwo *v1beta1.HTTPRoute = nil
+		deployment1           *appsv1.Deployment = nil
+		service1              *corev1.Service    = nil
+		deployment2           *appsv1.Deployment = nil
+		service2              *corev1.Service    = nil
+	)
+
+	var resourceCreationWaitTime = 30 * time.Second
+
+	Context("Create a HTTPRoute with backendref to service1, then update the HTTPRoute with backendref to service1 "+
+		"and service2, then update the HTTPRoute with backendref to just service2", func() {
+		It("Updates rules correctly with corresponding target groups after each update", func() {
```

Comment:
does it make sense to split "It" into multiple "It"s? it seems there are many things happen inside

Reply:
I agree, although this whole test relies on updating existing resources. The way our tests are written now, the resources will be cleaned after each test. We will need to address that first before I can split these out.